app/code/Magento/CatalogInventory/Helper/ProductStockIsQtyDecimal.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Magento\CatalogInventory\Helper;
+
+use Magento\CatalogInventory\Api\StockItemRepositoryInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class GetProductIsQtyDecimalService
+ */
+class ProductStockIsQtyDecimal
+{
+    /**
+     * @var StockItemRepositoryInterface
+     */
+    protected $stockItemRepository;
+
+    /**
+     * @param StockItemRepositoryInterface $stockItemRepository
+     *
+     * @return void
+     */
+    public function __construct(StockItemRepositoryInterface $stockItemRepository)
+    {
+        $this->stockItemRepository = $stockItemRepository;
+    }
+
+    /**
+     * @param $productId
+     *
+     * @return bool
+     */
+    public function execute($productId)
```

Comment:
`$productId` seems to be not the most descriptive name, as stock item repository get accepts stock item id